src/network/src/esp/include/NetworkManager.h

```diff
@@ -1,24 +1,28 @@
 #ifndef HIVE_CONNECT_NETWORKMANAGER_H
 #define HIVE_CONNECT_NETWORKMANAGER_H
 
+#include "INetworkManager.h"
+#include "TCPClient.h"
+#include "TCPServer.h"
 #include "logger/ILogger.h"
+#include "memory"
 #include <BaseTask.h>
 #include <Task.h>
+#include <optional>
 
 #include "esp_wifi.h"
 
 /**
- * @brief The network manager class. Handles the connnection to the network
+ * @brief The network manager class. Handles the connection to the network
```

Comment:
For now, it really only handles connecting and handling disconnection from the network. In the future, it will allow to store a list of the known agents in the swarm. The description will be updated as the features are developped.